node/core/runtime-api/src/lib.rs

```diff
@@ -36,22 +36,38 @@ use polkadot_primitives::v1::{Block, BlockId, Hash, ParachainHost};
 use sp_api::ProvideRuntimeApi;
 use sp_core::traits::SpawnNamed;
 
-use futures::prelude::*;
-use std::sync::Arc;
+use futures::{prelude::*, stream::FuturesUnordered, channel::oneshot, select};
+use std::{sync::Arc, collections::VecDeque, pin::Pin};
 
 const LOG_TARGET: &str = "runtime_api";
 
+/// The number of maximum parallel runtime api requests.
+const MAX_PARALLEL_REQUESTS: usize = 2;
```

Comment:
Is this intentionally very low? There are at least two locations I know from the top of my head that require to distinct chunks of information which could be perallalized (or already are) and would deplete this right away.